pettingzoo/classic/checkers/checkers.py

```diff
@@ -0,0 +1,608 @@
+# from __future__ import absolute_import, division, print_function
+from six.moves import range
+
+import itertools
+import copy
+import numpy as np
+import warnings
+
+from pettingzoo import AECEnv
+from gym import spaces
+from pettingzoo.utils.agent_selector import agent_selector
+from pettingzoo.utils import wrappers
+
+
+def env():
+    env = raw_env()
+    env = wrappers.TerminateIllegalWrapper(env, illegal_reward=-1)
+    env = wrappers.AssertOutOfBoundsWrapper(env)
+    env = wrappers.NaNRandomWrapper(env)
+    env = wrappers.OrderEnforcingWrapper(env)
+    return env
+
+
+class raw_env(AECEnv):
+
+    metadata = {"render.modes": ["human"]}
+
+    def __init__(self):
+        super().__init__()
+
+        self.ch = CheckersRules()
+        self.num_agents = 2
+        self.agents = ["player_{}".format(i) for i in range(self.num_agents)]
+        self.agent_order = list(self.agents)
+        self.last_turn = "black"
+
+        self._agent_selector = agent_selector(self.agent_order)
+
+        self.action_spaces = {name: spaces.Discrete(64 * 4) for name in self.agents}
+        self.observation_spaces = {
+            name: spaces.Box(low=0, high=1, shape=(8, 8, 4), dtype=np.int32)
+            for name in self.agents
+        }
+        self.observation = np.zeros((8, 8, 4))
+
+        self.reset()
+
+    def _read_observation(self):
+        # Use self.ch.flatboard to update self.observation
+        board = self.ch.flat_board()
+        obs = np.zeros((8, 8, 4))
+        for i, row in enumerate(board):
+            for j, sq in enumerate(row):
+                if sq > 0:
+                    obs[i, j, sq - 1] = 1
+        return np.array(obs)
+
+    def observe(self, agent):
+        return np.array(self.observation)
+
+    def seed(self, seed=None):
+        pass
+
+    def reset(self, observe=True):
+        self.ch = CheckersRules()
+        self.observation = self._read_observation()
+        self.num_moves_max = 300
+        self.num_moves = 0
+        self.agent_order = list(self.agents)
+        self.agent_selection = self._agent_selector.reset()
+        self.last_turn = "black"
+        self.rewards = {name: 0 for name in self.agents}
+        self.dones = {name: False for name in self.agents}
+        self.infos = {name: {"legal_moves": []} for name in self.agents}
+        self.infos[self.agent_selection]["legal_moves"] = self.legal_moves()
+        self.winner = -1
+        if observe:
+            return np.array(self.observation)
+
+    def _obs_rel_to_abs(self, pos):
+        # Convert location from (32) to (8, 8)
+        row = int(pos / 4)
+        pos64 = 0
+        if row % 2 == 0:
+            pos64 = 2 * pos + 1
+        else:
+            pos64 = 2 * pos
+        row = int(pos64 / 8)
+        col = int(pos64 % 8)
+        return (row, col)
+
+    def _obs_abs_to_rel(self, pos):

```

Comment:
is this function used anywhere?

Reply:
Also no, I guess I just wrote these out of convenience and ended up not needing them